engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/Aggregator.scala

```diff
@@ -12,7 +12,7 @@ import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
   - computeOutputType provides information for validation, completion etc.
   - we provide own versions of AggregateFunction methods to use type members instead of type parameters, as it makes it easier to use
  */
-trait Aggregator extends AggregateFunction[Any, Any, Any] {
+abstract class Aggregator extends AggregateFunction[AnyRef, AnyRef, AnyRef] {
```

Comment:
Yep, generally for better compatibility with java we should:
- be careful with abstract types
- ... with clastags/typetags
- use abstract class in "main" inheritance tree
- prefer methods instead of function members (`protected def foo: (T) => R`)
- avoid using package objects
- ...
... in all places where there is an option that we will use it from java.
WDYT about some developer guide instead writing down such comments in all those places? I can prepare such if you want.

Reply:
I would be nice I think